gcp/appengine/frontend_handlers.py

```diff
@@ -50,6 +51,26 @@ def check_rate_limit():
       abort(429)
 
 
+@blueprint.before_request
+def check_cors_preflight():
+  """Handle CORS preflight requests."""
+  if request.method != 'OPTIONS':
+    return None
+
+  response = make_response()
+  response.headers.add('Access-Control-Allow-Origin', 'http://localhost:8080')
+  response.headers.add('Access-Control-Allow-Methods', '*')
+  response.headers.add('Access-Control-Allow-Headers', '*')
+  return response
+
+
+@blueprint.after_request
+def add_cors_headers(response):
```

Comment:
where are these 2 functions called ?

Reply:
It's set up to be automatically called as part of `before_request` and `after_request`.